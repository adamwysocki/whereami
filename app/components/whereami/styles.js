'user strict';
/**
 * whereami component styles
 * @flow
 *
 */

 import { StyleSheet } from 'react-native';

 export default StyleSheet.create({
   container: {
     flex: 1,
     marginTop: 20,
     justifyContent: 'center',
     alignItems: 'center',
     backgroundColor: '#FFFFFF',
   },
   title: {
     color: '#ffffff',
     fontWeight: '900',
     marginTop: 10,
     fontSize: 18
   }
 });
