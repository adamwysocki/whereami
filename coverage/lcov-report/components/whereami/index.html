<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/whereami/</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">all files</a> components/whereami/
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>36/63</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.88% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">30% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/30</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="55" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 55%;"></div><div class="cover-empty" style="width:45%;"></div></div></td>
	<td data-value="55" class="pct medium">55%</td>
	<td data-value="60" class="abs medium">33/60</td>
	<td data-value="34.88" class="pct low">34.88%</td>
	<td data-value="43" class="abs low">15/43</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="20" class="abs low">6/20</td>
	<td data-value="32.14" class="pct low">32.14%</td>
	<td data-value="28" class="abs low">9/28</td>
	</tr>

<tr>
	<td class="file high" data-value="styles.js"><a href="styles.js.html">styles.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sat Aug 20 2016 11:12:22 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
