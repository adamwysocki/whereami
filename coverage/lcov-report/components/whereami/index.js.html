<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/whereami/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">all files</a> / <a href="index.html">components/whereami/</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">55% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>33/60</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.88% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">30% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">32.14% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/28</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101</td><td class="line-coverage quiet"><span class="cline-any cline-yes">5×</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-yes">4×</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1×</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">'user strict';<span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span>
/**
 * whereami component
 * @flow
 *
 * Google Places API Key AIzaSyCsy8VXr84z-pKP54gTc6X-PZtbTOBCj8A
 */
&nbsp;
import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Navigator
} from 'react-native';
&nbsp;
// import components
import Button from '../button';
import LocationContainer from '../locationContainer';
import styles from './styles';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
export default class whereami extends Component {
&nbsp;
  constructor(props) {<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
    super(props);
    // set some initial state
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      initialPosition: props.initialPosition || 'unknown',
      lastPosition: props.lastPosition || 'unknown',
      showing: props.showing || false,
      latitude: props.latitude || 0,
      longitude: props.longitu<span class="cstat-no" title="statement not covered" >de || 0</span>
    };
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  componentDidMount() {</span>
    // Load up the initial lat/lng
<span class="cstat-no" title="statement not covered" >    this._updateLocation</span>();
  }
&nbsp;
  // configure and return the top nav bar
<span class="fstat-no" title="function not covered" >  _getTopNavBar() {</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;Navigator.NavigationBar
        routeMapper={{
          LeftButton: (route, navigator, index, navState) =&gt; {<span class="fstat-no" title="function not covered" > },</span>
          RightButton: (route, navigator, index, navState) =&gt; { <span class="fstat-no" title="function not covered" >},</span>
          Title: (route, navigator, index, navState) =&gt;
<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >            { return (&lt;Text style={styles.title}&gt;Where Am I?&lt;/Text&gt;); },</span></span>
        }}
        style={{backgroundColor: 'gray'}}
        /&gt;
    );
  }
&nbsp;
  // use geolocation service to get lat/lng
<span class="fstat-no" title="function not covered" >  _updateLoc<span class="cstat-no" title="statement not covered" >ation</span>() {</span>
<span class="cstat-no" title="statement not covered" >    navigator.geolocation.getCurrentPosition(</span>
      (position) =<span class="fstat-no" title="function not covered" >&gt; {</span>
        var initialP<span class="cstat-no" title="statement not covered" >osition = JSON.stringify(</span>position);
<span class="cstat-no" title="statement not covered" >        this.setState({latitude: position.coords.latit</span>ude});
<span class="cstat-no" title="statement not covered" >        this.setState({longitude: position.coords.longit</span>ude});
<span class="cstat-no" title="statement not covered" >        this.setState({initialPosition});</span>
      },
      (error) =<span class="fstat-no" title="function not covered" >&gt;<span class="cstat-no" title="statement not covered" > alert(error.message),</span></span>
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
    );
  }
&nbsp;
  // toggle whether we're showing the location
<span class="fstat-no" title="function not covered" >  _wherePressed() {</span>
<span class="cstat-no" title="statement not covered" >    this.setState({showing: !this.state.showin</span>g});
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  render() {</span>
    var visibleComponent;
&nbsp;
    // if "showing" then show the location, otherwise initial view
<span class="cstat-no" title="statement not covered" >    if(this.state.showing) {</span>
<span class="cstat-no" title="statement not covered" >      visibleComponent = &lt;LocationContainer</span>
                          latitude={this.state.latitude}
                          longitude={this.state.longitude}
                          onPress={this._wherePressed.bind(this)}/&gt;
    } else {
<span class="cstat-no" title="statement not covered" >      visibleComponent = &lt;Button onPress={this._wherePressed.bind(this)}&gt;WHERE AM I?&lt;/Button&gt;</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;Navigator
        initialRoute={{ title: 'Main Scene', index: 0 }}
        renderScene={(route, navigator) =&gt;<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span>
          &lt;View style={styles.container}&gt;
            {visibleComponent}
          &lt;/View&gt;
        }
        navigationBar={this._getTopNavBar()}
      /&gt;
    );
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sat Aug 20 2016 10:26:30 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
