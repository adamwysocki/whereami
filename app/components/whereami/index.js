'use strict';
/**
 * whereami component
 * @flow
 *
 * Google Places API Key AIzaSyCsy8VXr84z-pKP54gTc6X-PZtbTOBCj8A
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Navigator
} from 'react-native';

// import components
import Button from '../button';
import LocationContainer from '../locationContainer';
import styles from './styles';

export default class whereami extends Component {

  constructor(props) {
    super(props);
    // set some initial state
    this.state = {
      initialPosition: props.initialPosition || 'unknown',
      lastPosition: props.lastPosition || 'unknown',
      showing: props.showing || false,
      latitude: props.latitude || 0,
      longitude: props.longitude || 0
    };
  }

  componentDidMount() {
    // Load up the initial lat/lng
    this._updateLocation();
  }

  // configure and return the top nav bar
  _getTopNavBar() {
    return (
      <Navigator.NavigationBar
        routeMapper={{
          LeftButton: (route, navigator, index, navState) => { },
          RightButton: (route, navigator, index, navState) => { },
          Title: (route, navigator, index, navState) =>
            { return (<Text style={styles.title}>Where Am I?</Text>); },
        }}
        style={{backgroundColor: 'gray'}}
        />
    );
  }

  // use geolocation service to get lat/lng
  _updateLocation() {
    navigator.geolocation.getCurrentPosition(
      (position) => {
        var initialPosition = JSON.stringify(position);
        this.setState({latitude: position.coords.latitude});
        this.setState({longitude: position.coords.longitude});
        this.setState({initialPosition});
      },
      (error) => alert(error.message),
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
    );
  }

  // toggle whether we're showing the location
  _wherePressed() {
    this.setState({showing: !this.state.showing});
  }

  render() {
    var visibleComponent;

    // if "showing" then show the location, otherwise initial view
    if(this.state.showing) {
      visibleComponent = <LocationContainer
                          latitude={this.state.latitude}
                          longitude={this.state.longitude}
                          onPress={this._wherePressed.bind(this)}/>
    } else {
      visibleComponent = <Button onPress={this._wherePressed.bind(this)}>WHERE AM I?</Button>
    }

    return (
      <Navigator
        initialRoute={{ title: 'Main Scene', index: 0 }}
        renderScene={(route, navigator) =>
          <View style={styles.container}>
            {visibleComponent}
          </View>
        }
        navigationBar={this._getTopNavBar()}
      />
    );
  }
}
