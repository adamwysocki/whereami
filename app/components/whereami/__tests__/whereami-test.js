// __tests__/whereami-test.js
import 'react-native';
import React from 'react';
import whereami from '../index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('whereami', () => {

  it('renders correctly initially', () => {
    const tree = renderer.create(
      <whereami />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
