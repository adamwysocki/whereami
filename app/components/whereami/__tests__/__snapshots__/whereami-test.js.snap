exports[`whereami renders correctly initially 1`] = `<whereami />`;
